Libraries/Microsoft.Teams.Plugins/Microsoft.Teams.Plugins.AspNetCore.BotBuilder/Controllers/MessageController.cs

```diff
@@ -0,0 +1,62 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Teams.Api.Auth;
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Teams.Apps.Extensions;
+using Microsoft.Teams.Api.Activities;
+using Microsoft.AspNetCore.Http;
+using System.Text.Json;
+
+namespace Microsoft.Teams.Plugins.AspNetCore.BotBuilder
+{
+    [ApiController]
+    public class MessageController : ControllerBase
+    {
+        private readonly IBotFrameworkHttpAdapter _adapter;
+        private readonly IBot _bot;
+
+        private readonly AspNetCorePlugin _plugin;
+
+        private readonly IHostApplicationLifetime _lifetime;
+
+        public MessageController(IBotFrameworkHttpAdapter adapter, IBot bot, AspNetCorePlugin plugin, IHostApplicationLifetime lifetime)
+        {
+            _plugin = plugin;
+            _lifetime = lifetime;
+            _adapter = adapter;
+            _bot = bot;
+        }
+
+        [HttpPost("/api/messages")]
+        public async Task<IResult> PostAsync()
+        {
+            HttpContext.Request.EnableBuffering();
```

Comment:
By default the request body can only be read once - for performance reasons. This and line 39 makes it so that it can be read multiple times - once by the BF adapter and once by our plugin.